packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -0,0 +1,3369 @@
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+          // BlockTracker
+          {
+            request: {
+              method: 'eth_blockNumber',
+              params: [],
+            },
+            response: {
+              result: '0x1',
+            },
+          },
+        ],
+      });
+
+      const selectedAddress = ETHERSCAN_TRANSACTION_BASE_MOCK.to;
+
+      const { networkController, transactionController } = await newController({
+        getSelectedAddress: () => selectedAddress,
+      });
+
+      const expectedLastFetchedBlockNumbers: Record<string, number> = {};
+      const expectedTransactions: Partial<TransactionMeta>[] = [];
+
+      const networkClients = networkController.getNetworkClientRegistry();
+      // NOTE(JL): This doesn't seem to work for the globally selected provider because of nock getting stacked on mainnet.infura.io twice
+      const networkClientIds = Object.keys(networkClients).filter(
+        (v) => v !== networkClientConfiguration.network,
+      );
+      await Promise.all(
+        networkClientIds.map(async (networkClientId) => {
+          const config = networkClients[networkClientId].configuration;
+          mockNetwork({
+            networkClientConfiguration: config,
+            mocks: [
+              // BlockTracker
+              {
+                request: {
+                  method: 'eth_blockNumber',
+                  params: [],
+                },
+                response: {
+                  result: '0x1',
+                },
+              },
+            ],
+          });
+          nock(getEtherscanApiHost(config.chainId))
+            .get(
+              `/api?module=account&address=${selectedAddress}&offset=40&sort=desc&action=txlist&tag=latest&page=1`,
+            )
+            .reply(200, ETHERSCAN_TRANSACTION_RESPONSE_MOCK);
+
+          transactionController.updateIncomingTransactions([networkClientId]);
+
+          expectedLastFetchedBlockNumbers[
+            `${config.chainId}#${selectedAddress}#normal`
+          ] = parseInt(ETHERSCAN_TRANSACTION_BASE_MOCK.blockNumber, 10);
+          expectedTransactions.push({
+            blockNumber: ETHERSCAN_TRANSACTION_BASE_MOCK.blockNumber,
+            chainId: config.chainId,
+            type: TransactionType.incoming,
+            verifiedOnBlockchain: false,
+            status: TransactionStatus.confirmed,
+          });
+          expectedTransactions.push({
+            blockNumber: ETHERSCAN_TRANSACTION_BASE_MOCK.blockNumber,
+            chainId: config.chainId,
+            type: TransactionType.incoming,
+            verifiedOnBlockchain: false,
+            status: TransactionStatus.failed,
+          });
+        }),
+      );
+
+      // we have to wait for the mutex is released after the 5 second gap between API calls finishes
```

Comment:
worded a little weird

Reply:
I can improve this

Reply:
https://github.com/MetaMask/core/pull/3643/commits/eb302780690043b0e7cbe46219d23770db600856